lookout/style/typos/token_parser.py

```diff
@@ -0,0 +1,162 @@
+import re
+from collections import defaultdict, deque
+
+import Stemmer
+
+
+def uast2sequence(root):
+    sequence = []
+    nodes = defaultdict(deque)
+    stack = [root]
+    nodes[id(root)].extend(root.children)
+    while stack:
+        if nodes[id(stack[-1])]:
+            child = nodes[id(stack[-1])].popleft()
+            nodes[id(child)].extend(child.children)
+            stack.append(child)
+        else:
+            sequence.append(stack.pop())
+    return sequence
+
+
+class TokenParser:
+    """
+    Common utilities for splitting and stemming tokens.
+    """
+    NAME_BREAKUP_RE = re.compile(r"[^a-zA-Z]+")  #: Regexp to split
+    # source code identifiers.
+    STEM_THRESHOLD = 6  #: We do not stem splitted parts shorter than
+    # or equal to this size.
+    MAX_TOKEN_LENGTH = 256  #: We cut identifiers longer than this value.
+    MIN_SPLIT_LENGTH = 3  #: We do not split source code identifiers shorter
+    # than this value.
+
+    def __init__(self, stem_threshold=STEM_THRESHOLD,
+                 max_token_length=MAX_TOKEN_LENGTH,
+                 min_split_length=MIN_SPLIT_LENGTH):
+        self._stemmer = Stemmer.Stemmer("english")
+        self._stemmer.maxCacheSize = 0
+        self._stem_threshold = stem_threshold
+        self._max_token_length = max_token_length
+        self._min_split_length = min_split_length
+
+    @property
+    def stem_threshold(self):
+        return self._stem_threshold
+
+    @stem_threshold.setter
+    def stem_threshold(self, value):
+        if not isinstance(value, int):
+            raise TypeError(
+                "stem_threshold must be an integer - got %s" %
+                type(value))
+        if value < 1:
+            raise ValueError(
+                "stem_threshold must be greater than 0 - got %d" %
+                value)
+        self._stem_threshold = value
```

Comment:
incorrect logs (same below)

Reply:
This is a copy past from src-d/ml, didn't write this code. There were problems with protobuf, and this quick solution was suggested to me. Will try proper import, should work already.

Reply:
Oh ok! Still I think it might be nice to have logs that do match the tests performed (so must be lesser than 1 instead of greater than 0). It might be nice to PR that to ml also if this mismatch is also present there.

Reply:
yeap, better to do a proper import.